### DISCUSSION TITLE
List tags

### DISCUSSION THREAD

**Discussion by haf** [swellstores/community#99] [Help] [answered]
Hi,

How can I, with the API, list all product tags? Is there a similar endpoint, like for attributes that doesn't require me to paginate all products?

Kind regards,
Henrik

**Comment by rodrigoddalmeida** [comment, marked as answer]
To fetch a list of all product tags, you can query the `/products/:group` endpoint with an aggregation like this: 

```
{
  "aggregate": [
    {
      "$unwind": "$tags"
    },
    {
      "$group": {
        "_id": null,
        "allTags": {
          "$addToSet": "$tags"
        }
      }
    },
    { "$project": { "_id": 0, "allTags": 1 }}
  ]
}
```

The `tags` field in our base Product model is a simple list of strings, which may not be the best fit if you want a more robust reference between tags or tags -> products - that may work better with categories and attributes.

**Reply by haf** [reply]
How would I model it better? I can't use categories because that's already defining the product type; now I want to attach a list of string / list of metadata objects to each product.

**Comment by haf** [comment]
When I try this answer, this is what I receive:

<img width="859" alt="image" src="https://user-images.githubusercontent.com/193115/220197704-b5bf4130-e434-4cd7-be3a-1099c244fb03.png">
